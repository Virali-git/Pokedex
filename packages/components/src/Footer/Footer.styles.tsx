export const appBarStyle = {
  backgroundColor: " #ff6600",
  top: "auto",
  bottom: 0,
  color: "#000",
  boxShadow: "0px 2px 5px rgba(0, 0, 0, 0.2)",
};

export const ContainerStyle = {
  display: "flex",
  justifyContent: "space-between",
  alignItems: "center",
  padding: "20px",
};
