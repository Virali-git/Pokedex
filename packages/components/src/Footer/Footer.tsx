import React from "react";
import { styled } from "@mui/material/styles";
import {
  AppBar,
  Typography,
  Container,
  IconButton,
} from "@mui/material";
import FacebookIcon from "@mui/icons-material/Facebook";
import InstagramIcon from "@mui/icons-material/Instagram";
import LinkedInIcon from "@mui/icons-material/LinkedIn";

type IProps = { data: string };

const StyledAppBar = styled(AppBar)(({ theme }) => ({
  backgroundColor: " #ff6600", 
  top: "auto",
  bottom: 0,
  color: "#000",
  boxShadow: "0px 2px 5px rgba(0, 0, 0, 0.2)",
}));
export const Footer: React.FC<IProps> = ({ data }) => {
  return (
    <StyledAppBar
      position="static"
    >
      <Container
        maxWidth="lg"
        sx={{
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center",
          padding: "20px",
        }}
      >
        <Typography variant="body1" sx={{ fontWeight: "bold" }}>
          All rights reserved
        </Typography>
        <div style={{ display: "flex", gap: "5px" }}>
          <IconButton color="inherit" aria-label="Facebook">
            <FacebookIcon />
          </IconButton>
          <IconButton color="inherit" aria-label="Instagram">
            <InstagramIcon />
          </IconButton>
          <IconButton color="inherit" aria-label="LinkedIn">
            <LinkedInIcon />
          </IconButton>
        </div>
      </Container>
    </StyledAppBar>
  );
};
