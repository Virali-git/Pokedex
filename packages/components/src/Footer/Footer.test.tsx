import React from "react";
import { render, screen } from "@testing-library/react";
import { Footer } from "./Footer";

describe("Footer Component", () => {
  it("renders the 'All rights reserved' text", () => {
    render(<Footer data="Test Data" />);
    const allRightsReservedText = screen.getByText(/All rights reserved/i);
    expect(allRightsReservedText).toBeInTheDocument();
  });

  it("renders the social media icons", () => {
    const {getByText} =render(<Footer data="Test Data" />);
    const facebookIcon = screen.getByLabelText("Facebook");
    const instagramIcon = screen.getByLabelText("Instagram");
    const linkedInIcon = screen.getByLabelText("LinkedIn");

    expect(facebookIcon).toBeInTheDocument();
    expect(instagramIcon).toBeInTheDocument();
    expect(linkedInIcon).toBeInTheDocument();
    expect(getByText).toMatchSnapshot();
  });
});
