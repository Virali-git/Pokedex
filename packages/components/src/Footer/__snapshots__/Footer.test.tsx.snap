// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders the social media icons 1`] = `[Function]`;
